databricks_cli/stack/api.py

```diff
@@ -234,6 +245,58 @@ def _update_job(self, job_settings, job_id):
         click.echo("Updating Job")
         self.jobs_client.reset_job({'job_id': job_id, 'new_settings': job_settings})
 
+    def _deploy_workspace(self, resource_properties, physical_id, overwrite):
+        """
+        Deploy workspace asset.
+
+        :param resource_properties: dict of properties for the workspace asset. Must contain the
+        'source_path' and 'path' fields. The other fields will be inferred if not provided.
+        :param physical_id: dict containing physical identifier of workspace asset on databricks.
+        Should contain the field 'path'.
+        :param overwrite: Whether or not to overwrite the contents of workspace notebooks.
+        :return: (dict, dict) of (physical_id, deploy_output). physical_id is the physical ID for
+        the stack status that contains the workspace path of the notebook or directory on datbricks.
+        deploy_output. Is the initial information about the asset on databricks at deploy time.
+        """
+        try:
+            local_path = resource_properties['source_path']
+            workspace_path = resource_properties['path']
+        except KeyError as e:
+            raise StackError("%s doesn't exist in workspace resource properties" % str(e))
+
+        lang_fmt = WorkspaceLanguage.to_language_and_format(local_path)  # Guess language and format
+        language, fmt = None, None
+        if lang_fmt:
+            language, fmt = lang_fmt
+
+        # Plug in optional properties.
+        if 'language' in resource_properties:
+            language = resource_properties['language']
+        if 'format' in resource_properties:
+            fmt = resource_properties['format']
+
+        object_type = "DIRECTORY" if os.path.isdir(local_path) else "NOTEBOOK"
+        if 'object_type' in resource_properties:
+            object_type = resource_properties['object_type']
+
+        click.echo('sync {} {} to {}'.format(object_type, local_path, workspace_path))
+        if object_type == 'NOTEBOOK':
+            self.workspace_client.mkdirs(
+                os.path.dirname(workspace_path))  # Make directory in workspace if not exist
+            self.workspace_client.import_workspace(local_path, workspace_path, language, fmt,
+                                                   overwrite)
+        elif object_type == 'DIRECTORY':
+            self.workspace_client.import_workspace_dir(local_path, workspace_path, overwrite,
+                                                       exclude_hidden_files=True)
+
+        if physical_id and physical_id['path'] != workspace_path:
+            # Alert if last deployment's path differs from this one.
+            click.echo("Workspace asset had path changed from {} to {}".format(physical_id['path'],
+                                                                               workspace_path))
+        deploy_output = self.workspace_client.get_status_json(workspace_path)
+
+        return {'path': workspace_path}, deploy_output
```

Comment:
assign `{'path': workspace_path}` to a variable `new_physical_id` to improve readability

Reply:
👍